packages/theme/src/theme.types.ts

```diff
@@ -42,14 +42,21 @@ export interface ComponentDefaultProps
   extends Omit<ThemingProps, "styleConfig">,
     Dict {}
 
-export type ThemingPropsThunk<T> =
-  | T
-  | ((
-      props: Dict &
-        Omit<ThemingProps, "styleConfig"> & {
-          colorMode: ColorMode
-        },
-    ) => T)
+export interface ThemeComponentProps<Theme extends ChakraTheme = ChakraTheme>
+  extends Dict,
+    Omit<ThemingProps, "styleConfig"> {
```

Comment:
Could this extend `ComponentDefaultProps` instead to simplify?

Reply:
If we take for granted that every default props are passed to the function, then yes it could.

Good catch @with-heart